sources/packages/backend/libs/integrations/src/atbc-integration/atbc-integration.processing.service.ts

```diff
@@ -56,52 +61,80 @@ export class ATBCIntegrationProcessingService {
+    const studentDisabilityUpdates =
+      await this.atbcService.getStudentDisabilityStatusUpdatesByDate();
 
+    const studentsToUpdate: StudentDisabilityStatusDetail[] =
+      studentDisabilityUpdates.map((student) => ({
+        sin: student.SIN,
+        lastName: student.APP_LAST_NAME,
+        birthDate: getDateOnlyFromFormat(student.BIRTH_DTE, ATBC_DATE_FORMAT),
+        disabilityStatus: student.D8Y_TYPE as DisabilityStatus,
+        disabilityStatusUpdatedDate: getDateOnlyFromFormat(
+          student.D8Y_DTE,
+          ATBC_DATE_FORMAT,
+        ),
+      }));
+    let updatedDisabilityStatusCount = 0;
+    this.logger.log(
+      `Processing disability status requests for ${studentsToUpdate.length} students`,
+    );
+    processSummaryResult.summary.push(
+      `Total disability status requests processed: ${studentsToUpdate.length}`,
+    );
+    if (studentsToUpdate.length) {
       const processingResults = await processInParallel(
-        (student: Student) => this.processStudentPDStatus(student),
-        students,
+        (studentDisabilityStatusDetail: StudentDisabilityStatusDetail) =>
+          this.processStudentDisabilityStatusUpdate(
+            studentDisabilityStatusDetail,
+          ),
+        studentsToUpdate,
       );
 
-      updatedPDstatusCount += processingResults.filter(
+      updatedDisabilityStatusCount += processingResults.filter(
         (result) => result,
       ).length;
     }
-    return {
-      pdRequestsProcessed: students.length,
-      pdRequestsUpdated: updatedPDstatusCount,
-    };
+    processSummaryResult.summary.push(
+      `Students updated with disability status: ${updatedDisabilityStatusCount}`,
+    );
+    return processSummaryResult;
   }
 
   /**
-   * Check the PD status of given student at ATBC and update the
-   * status in db if the status is updated at ATBC end.
-   * @param student student.
+   * Update the disability status of the given student.
+   * Identify the student by SIN, last name and birth date
+   * and update the disability status of the student only if the status has changed.
+   * e.g. if ATBC response has same disability status (PD/PPD) which is already present
+   * in the system, update does not happen.
+   * @param studentDisabilityDetail student disability status details.
    */
-  private async processStudentPDStatus(student: Student): Promise<boolean> {
-    const atbcPDCheckResponse = await this.atbcService.checkStudentPDStatus({
-      id: student.id,
-      sin: student.sinValidation.sin,
-    });
-
+  private async processStudentDisabilityStatusUpdate(
+    studentDisabilityStatusDetail: StudentDisabilityStatusDetail,
+  ): Promise<boolean> {
+    const student =
+      await this.studentService.getStudentBySINAndLastNameAndBirthDate(
+        studentDisabilityStatusDetail.sin,
+        studentDisabilityStatusDetail.lastName,
+        getISODateOnlyString(studentDisabilityStatusDetail.birthDate),
```

Comment:
just double checking, can't we use the date in typeorm query?

Reply:
I can double check that. typeorm column property is string(expects iso string) and so I converted.

Reply:
as birthDate is string, it doesn't allow me to use date object in typeorm query.

![image](https://github.com/bcgov/SIMS/assets/54600590/c93e7676-f0b6-481f-86f7-a45a687ee58a)

Reply:
Thanks @dheepak-aot  for checking